app/javascript/packages/document-capture/components/file-image.jsx

```diff
@@ -13,12 +14,17 @@ import useIfStillMounted from '../hooks/use-if-still-mounted';
  * @param {FileImageProps} props Props object.
  */
 function FileImage({ file, alt, className }) {
-  const [imageData, setImageData] = useState(/** @type {string?} */ (null));
+  const cache = useContext(FileBase64CacheContext);
+  const [, forceRender] = useState((prevState = 0) => 1 - prevState);
```

Comment:
I need to double-check, but I'm pretty sure I got this backwards between initializer and setter where the "magic" of force rendering should happen. IIRC, the `useState` callback here would only be called once at initialization.

https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

> If the initial state is the result of an expensive computation, you may provide a function instead, which will be executed only on the initial render:

Instead, I think it should be:

```js
const [, forceRender ] = useState();
// ...
forceRender((prevState = 0) => 1 - prevState);
```

That being said, the fact that it appeared to be working correctly in testing is puzzling 🤔

Reply:
> That being said, the fact that it appeared to be working correctly in testing is puzzling 🤔

What's happening is roughly:

```js
const [/* value */, forceRender] = useState((prevState = 0) => 1 - prevState);
// `value` is `1`

// ...
ifStillMounted(forceRender)();
// `value` is now `undefined`, because above is equivalent to `forceRender(undefined)`.
// Because state has changed, the component rerenders.
```

I think it'd be possible to write a test for this by checking the case that `file` prop can change.